wezterm-gui/src/termwindow/mod.rs

```diff
@@ -1209,6 +1209,13 @@ impl TermWindow {
                 } => {
                     self.update_title();
                 }
+                MuxNotification::Alert {
+                    alert: Alert::MouseCursorShapeChanged,
+                    pane_id,
+                } => {
+                    // FIXME: Is there a cache that needs to be invalidated like for PaletteChanged?
```

Comment:
There isn't an explicit cache, but you will need to cause the logic in the normal mouse event processing flow to trigger and assess what the cursor should be here, based on its position.

I don't have a great suggestion for this off the top of my head. Most likely will need to add a helper function to figure out if the mouse cursor is over the terminal area and if so, get the shape from the pane identified by `pane_id` and apply it to the window.